src_cpp/opennnBridge/nerlWorkerOpenNN.cpp

```diff
@@ -87,27 +96,28 @@ namespace nerlnet
             case MODEL_TYPE_AE_CLASSIFIER: // Get Loss Values , Add RED support - David's Thesis
             {
                 std::shared_ptr<opennn::NeuralNetwork> neural_network = get_neural_network_ptr();
-                Index num_of_samples = _aec_data_set->dimension(0);
+                int num_of_samples = (int)_aec_data_set->dimension(0);
                 Index num_of_labels = 1;
                 Index inputs_number = neural_network->get_inputs_number();
                 Tensor<Index, 1> inputs_dimensions(2);
                 inputs_dimensions.setValues({num_of_samples, inputs_number});
                 fTensor2DPtr results = std::make_shared<fTensor2D>(num_of_samples, num_of_labels + num_of_labels*2); // LOWER/UPPER for each label
                 fTensor2DPtr calculate_res = std::make_shared<fTensor2D>(num_of_samples, neural_network->get_outputs_number());
                 *calculate_res = neural_network->calculate_outputs(TrainData->data(), inputs_dimensions);
-
-                // MSE Calculation
                 fTensor2DPtr loss_values_mse = std::make_shared<fTensor2D>(num_of_samples, 1);
+                fTensor2DPtr loss_values_return = std::make_shared<fTensor2D>(num_of_samples, 1);
                 fTensor2D diff = (*calculate_res - *_aec_data_set);
                 fTensor2D squared_diff = diff.pow(2);
                 fTensor1D sum_squared_diff = squared_diff.sum(Eigen::array<int, 1>({1}));
                 fTensor1D mse1D = (1.0 / static_cast<float>(_aec_data_set->dimension(0))) * sum_squared_diff;
-                fTensor2D mse2D = mse1D.reshape(Eigen::array<int, 2>({num_of_samples, 1}));
+                fTensor2D mse2D = mse1D.reshape(Eigen::array<int, 2>({(int)num_of_samples, 1}));
+                // cout << "MSE2D: " << mse2D << endl;
                 *loss_values_mse = mse2D;
```

Comment:
This can cause a memory issue. You try to put a tensor that will be deleted when this skope ends into a shared pointer. The shared pointer cannot take control of this memory block. This is a wrong usage of shared pointers.